flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/RelTimeIndicatorConverter.java

```diff
@@ -544,11 +595,110 @@ private RelNode convertAggInput(Aggregate agg) {
+                                newAlias, newFieldRef, window.size(), window.slide());
+            } else if (oldWindow instanceof SessionGroupWindow) {
+                SessionGroupWindow window = (SessionGroupWindow) oldWindow;
+                newWindow = new SessionGroupWindow(newAlias, newFieldRef, window.gap());
+            } else {
+                throw new TableException(
+                        String.format(
+                                "This is a bug and should not happen. Please file an issue. Invalid window %s.",
+                                oldWindow.getClass().getSimpleName()));
+            }
+            List<PlannerNamedWindowProperty> newNamedPropertiesList =
+                    JavaConverters.seqAsJavaListConverter(oldNamedProperties).asJava().stream()
+                            .map(
+                                    namedProperty -> {
+                                        if (namedProperty.getProperty()
+                                                instanceof PlannerRowtimeAttribute) {
+                                            return new PlannerNamedWindowProperty(
+                                                    namedProperty.getName(),
+                                                    new PlannerRowtimeAttribute(newAlias));
+                                        } else {
+                                            return namedProperty;
+                                        }
+                                    })
+                            .collect(Collectors.toList());
+            newNamedProperties =
+                    JavaConverters.iterableAsScalaIterableConverter(newNamedPropertiesList)
+                            .asScala()
+                            .toSeq();
+        } else {
+            newWindow = oldWindow;
+            newNamedProperties = oldNamedProperties;
+        }
+        return new FlinkLogicalWindowAggregate(
+                agg.getCluster(),
+                agg.getTraitSet(),
+                newInput,
+                agg.getGroupSet(),
+                updatedAggCalls,
+                newWindow,
+                newNamedProperties);
+    }
+
+    private RelNode visitInvalidRel(RelNode node) {
+        throw new TableException(
+                String.format(
+                        "This is a bug and should not happen. Please file an issue. Unknown node %s.",
+                        node.getRelTypeName()));
+    }
+
     // ----------------------------------------------------------------------------------------
     //                                       Utility
     // ----------------------------------------------------------------------------------------
 
     private RelNode materializeProcTime(RelNode node) {
+        // If input is empty values, ignore materialize
+        if (node instanceof FlinkLogicalValues
+                && FlinkLogicalValues.isEmpty((FlinkLogicalValues) node)) {
+            return node;
+        }
```

Comment:
Does this really need ? the following code can handle `empty values`

Reply:
Actual there is no need to do, but if I don't add the logical, there are existed UT xml changed.
So I add the logical only to avoid the existed UT xml changed.

Reply:
In the original version, after time indicator materialize, PRUNE_EMPTY_RULES would prune project with empty values as input node.
After the refactor, if we add a calc to convert proc_time empty values, there is no rules to reduce this new added calc.

Reply:
effected UT including : LimitTest.testLimit0WithProcessingTime, LimitTest.testLimit0WithProcessingTimeDesc, LimitTest.testLimit0WithProcessingTimeDescSecond, LimitTest.testLimit0WithProcessingTimeSecond, LimitTest.testLimit0WithoutOffset

Reply:
please add some comments to explain why we the empty values ignore materialize in the code ?